### PR TITLE
[libjpeg-turbo] Update the version to 3.0.0

### LINKED ISSUE #15731: [libjpeg-turbo] renames libraries, missing vcpkg_fixup_pkgconfig
**Describe the bug**
The port currently renames the upstream library names.
Furthermore *.pc files are installed without calling vcpkg_fixup_pkgconfig.

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/15731
1. Update version to 3.0.0.
2. Fix upstream renaming of libraries.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

All features passed with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

### CODE REVIEW

**Review by dg0yt — CHANGES_REQUESTED** [review]
Do not remove `usage`. The discussions which lead to its current wording showed that it must be explicit.

File: ports/libjpeg-turbo/workaround_cmake_system_processor.patch [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
--string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} CMAKE_SYSTEM_PROCESSOR_LC)
-+string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" CMAKE_SYSTEM_PROCESSOR_LC)
- set(COUNT 1)
- foreach(ARCH ${CMAKE_OSX_ARCHITECTURES})
-   if(COUNT GREATER 1)
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index efd101a..c289358 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -94,7 +94,7 @@ include(cmakescripts/PackageInfo.cmake)
```
Comment (dg0yt) [inline ports/libjpeg-turbo/workaround_cmake_system_processor.patch:5]: No need to update this patch if it applies cleanly.

**Comment by dg0yt** [comment]
And please remove any reference to "the modification of the library name" - there is no such modification at the moment.

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/libjpeg-turbo/usage [hunk 2]
```diff
@@ -6,4 +6,4 @@ libjpeg-turbo is compatible with built-in implementation-agnostic CMake targets:
 libjpeg-turbo provides CMake targets for the TurboJPEG C API:
 
     find_package(libjpeg-turbo CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:libjpeg-turbo::turbojpeg>,libjpeg-turbo::turbojpeg,libjpeg-turbo::turbojpeg-static>)
+    target_link_libraries(main PRIVATE libjpeg-turbo::jpeg libjpeg-turbo::turbojpeg)
```
Comment (dg0yt) [inline ports/libjpeg-turbo/usage:9]: Why did you add `libjpeg-turbo::jpeg` here? I think that this target doesn't belong to the direct link libraries *for the TurboJPEG C API*. It was deliberately omitted in the original version of `usage`.

And why did you change it at all? On x64-osx, I still get `libjpeg-turbo::turbojpeg-static`. 
~~~
% grep -i add_lib packages/libjpeg-turbo_x64-osx/share/libjpeg-turbo/*cmake
packages/libjpeg-turbo_x64-osx/share/libjpeg-turbo/libjpeg-turboTargets.cmake:add_library(libjpeg-turbo::turbojpeg-static STATIC IMPORTED)
packages/libjpeg-turbo_x64-osx/share/libjpeg-turbo/libjpeg-turboTargets.cmake:add_library(libjpeg-turbo::jpeg-static STATIC IMPORTED)
~~~

So `usage` is broken now.

Comment (jimwang118) [reply ports/libjpeg-turbo/usage:9]: After updating the version, the usage generated by the upstream is the updated usage. The test is normal under Windows, but I did not test it under `osx` and `Linux`, so this error was caused. I will restore the original usage.

**Pull request merged**